### ISSUE TITLE
Define supported Go versions

### ISSUE THREAD

**Issue by kzys** [firecracker-microvm/firecracker-go-sdk#364]
The SDK currently doesn't have a policy regarding supported Go versions. We have been supporting Go 1.11 for a while even Go team doesn't support the version.

Right now, the motivation tends to come from dependencies. If they are having vulnerabilities and we want to upgrade them, we tend to realize that newer dependencies don't support older Go versions.

We probably should have a version supporting policy upfront, instead of deciding case-by-case basis. If supporting old Go versions have some value, we may be okay to ask our dependencies to support them.

**Comment by ginglis13** [comment]
Seeing this on other recent PRs (e.g. #440). Looks like our dependency `ltag` does not support Go <= 1.15:
```
GO111MODULE=off GOBIN=/home/runner/work/firecracker-go-sdk/firecracker-go-sdk/testdata \
go get github.com/kunalkushwaha/ltag
# github.com/kunalkushwaha/ltag
Error: ../../../go/src/github.com/kunalkushwaha/ltag/bash.go:25:15: undefined: os.ReadFile
Error: ../../../go/src/github.com/kunalkushwaha/ltag/dockerfile.go:23:15: undefined: os.ReadFile
Error: ../../../go/src/github.com/kunalkushwaha/ltag/golang.go:23:15: undefined: os.ReadFile
Error: ../../../go/src/github.com/kunalkushwaha/ltag/makefile.go:15:15: undefined: os.ReadFile
make: *** [Makefile:125: testdata/ltag] Error 2
Error: Process completed with exit code 2.
```
`os.ReadFile` was added in Go 1.16

**Comment by Joffref** [comment]
Maybe, version of the ltag binary must be setted using `go get github.com/kunalkushwaha/ltag@version` in the [Makefile](https://github.com/firecracker-microvm/firecracker-go-sdk/blob/dd9dd9395c36b19cff8000a9c3e72b772805f005/Makefile#L124-L127) to provides two versions, one for 1.15 <= and one for 1.16 >=  ?

**Comment by austinvazquez** [comment]
To me the way is to drop support for Go <= v1.16. v1.17 just ended support in August 2022 and users may still be in transition.

Also we should pin the dependency versions for reproducible builds but that is a separate issue.

**Comment by Kern--** [comment]
As @kzys mentioned, that would require users on Debian 11 or Ubuntu 20.04 to download go outside of their normal package managers. Debian 11 just went stable last year and is supported for [4 more years](https://www.debian.org/News/2021/20210814), and Ubuntu 20.04 is still supported for [2 years](https://ubuntu.com/about/release-cycle).

I don't think it's unreasonable to say we will only support go versions that are still supported upstream, but I don't think that's the only consideration.

**Comment by austinvazquez** [comment]
That's fair I wasn't viewing support from the linux distribution pov. Now I'm conflicted again. I want to consider those users, but 4 more years of Go 1.15 support 👀

**Comment by Joffref** [comment]
I agree with @Kern-- as a Firecracker go sdk user these "feature" is quite important to me as I'm working in constrained environments.

**Comment by kzys** [comment]
I've updated my comment above. Dropping Go 1.15 and 1.17 support may be a bummer for Ubuntu 18.04 LTS users, but other distros seem fine.

https://github.com/firecracker-microvm/firecracker-go-sdk/issues/364#issuecomment-942535398
